sherpa/astro/tests/test_astro_data2.py

```diff
@@ -4205,3 +4202,162 @@ def test_pha_ungroup_background_remove(caplog):
     assert not bkg.grouped
 
     assert len(caplog.record_tuples) == 0
+
+
+def test_pha_check_background_ids_basic():
+    """Check background_ids can be used.
+
+    This is a basic run through to check the behavior.
+    """
+
+    pha = DataPHA("src", [1, 2, 3], [1, 1, 1])
+    b1 = DataPHA("b1", [1, 2, 3], [1, 1, 1])
+    b2 = DataPHA("b2", [1, 2, 3], [1, 1, 1])
+
+    assert len(pha.background_ids) == 0
+
+    pha.set_background(b1)
+    assert pha.background_ids == pytest.approx([1])
+
+    pha.set_background(b2, id="up")
+    assert pha.background_ids == pytest.approx([1, "up"])
+
+    pha.delete_background(1)
+    assert pha.background_ids == pytest.approx(["up"])
+
+    # Check we can delete a background by setting background_ids
+    #
+    pha.background_ids = []
+    assert pha.background_ids == []
+
+    # Does the order matter?
+    #
+    pha.set_background(b2, id="up")
+    pha.set_background(b1)
+    assert pha.background_ids == pytest.approx(["up", 1])
+
+    # Remove one element.
+    #
+    pha.background_ids = [1]
+    assert pha.background_ids == pytest.approx([1])
+
+    # We can do the following, which is technically a no-op but may
+    # change some internal state. This also tests using an
+    # iterable-that-is-not-a-list.
```

Comment:
I would argue that iterable-that-is-not-a-list is not part of the intended API. If it works, great, but I don't think we have to guarantee that for the future.

Reply:
This at least gives us a canary if it does change. Depending on my mood I get annoyed with or approve wholeheartedly of the the Python duck-typing "hey, you can use this as it's an iterable" approach.